Add KDE support to start_jupyter_cm and update documentation

jat255 opened the pull request:
Added support for KDE to add context menu entries for the Dolphin file browser. I tried to stay as close to the existing structure for the Gnome script as possible, but some changes were necessary (needed to create .desktop file, for instance). Happy to take any suggestions for improvement.

jat255 commented on start_jupyter_cm/kde.py:
```diff
@@ -0,0 +1,137 @@
+        Either 'notebook' or 'qtconsole'
+    """
+    if terminal == 'qtconsole':
+        terminal += '-py2'
```
On my system, this is necessary to get hyperspy in a python2 console. This should be changed, or another option should be added once we move to python3.

francisco-dlp commented:
This is awesome! I'm currently busy with the release of HyperSpy 0.8.3, but I'll review this asap.

sem-geologist commented:
Wouldn't it be better to propose this function straight in jupyter development?
Also by default KDE have terminal available on the right click, and launching the jupyter notebook at the wanted directory is really just few sec away from there. Is the ~150 line of code worth it? I am asking this as KDE user. I understand the shortcut under windows which have terrible cmd interface and navigating to the wanted directory is a nightmare. But on the linux, we have no such problem.

Another peculiarity or even disadvantage I see is this excerpt from the README

> Note that in GNOME and KDE, the processes run in the background. As of Jupyter
  Notebook 4.1 there is no way to shutdown the server from the notebook
  UI. To stop the server one has to manually kill the process.  

maybe better approach would be to launch it with terminal, so that there would be possibility to shutdown the jupyter server gracefully?

ericpre commented:
@jat255: could you please solve the merge conflict and update to the use the correct python executable?
I guess it would be good to add support for conda environment, similarly to the way it is done in #8.

jat255 commented:
Ok, this should be ready for review again. To help prevent duplicate code, I changed the `gnome.py` to `linux.py`, and added KDE support to it. The code detects if the KDE/Nautilus paths are present, and uses that to determine what code to run. 

KDE requires full .desktop files to add things to the service menu, so that code has been added. I also cleaned up some of the strings and such. I have both KDE and Gnome installed, and it appears to work for me, but I would appreciate some additional testing.

jat255 commented:
Oh, I just realized we claim to support down to v3.5, and I used f-strings throughout. I suppose I should change those to `str.format()`

jat255 commented:
Ok, f-strings removed.

jat255 commented on start_jupyter_cm/linux.py:
```diff
@@ -0,0 +1,170 @@
+         "Exec='{0}/.local/share/kservices5/ServiceMenus/"
+         "Jupyter_{1}_here{2}' %f\n"
+         "Icon={3}\n"
+         "Name=Jupyter {4}{2} here\n"
```
Yup, good call; that was unintentional. I'll update it shortly.

francisco-dlp replied:
This little issue is a good example of why we should embrace f-strings. Why not reverting to using f-string and removing support for 3.5 at this opportunity?

jat255 replied:
:+1: I didn't want to make that decision, but if we're okay removing 3.5, then I'd be happy to swap back to f-strings

ericpre replied:
I am fine with both approaches and I agree that it is not a big deal to drop support for python 3.3, 3.4 and 3.5.

francisco-dlp replied:
Then let's drop support for Python < 3.6. f-string makes the code mode readable, so I think that it'll be good to do the same for the rest of the project in the hyperspy organization.

ericpre replied:
Agree, HyperSpy test suite is currently broken on python 3.5 because of installation issue in conda environment; I have sent a PR with more details to drop python 3.5: https://github.com/hyperspy/hyperspy/pull/2220

ericpre commented:
I have with dolphin installed (Mate desktop) and the service menu folder `/usr/share/kservices5/ServiceMenus` and `~/.local/share/kservices5/ServiceMenus` are not present. A reason may be it is not a KDE desktop environment but still I would hope that it should work for dolphin regardless of the desktop environment. IThese issues should be fixed in https://github.com/jat255/start_jupyter_cm/pull/1.

ericpre commented:
@jat255, just poking you on this PR and https://github.com/jat255/start_jupyter_cm/pull/1! ;)

jat255 commented:
Closed in lieu of #20
